marketplace_builder/views/pages/reference/form-configurations/validation.liquid

```diff
@@ -0,0 +1,246 @@
+
+{% raw %}
+<pre><code class="language-yaml">
+            rate_amount:
+              validation:
+                presence: true
+                numericality:
+                  greater_than_or_equal_to: 0
+</code></pre>
+{% endraw %}
+
+Available options are:
+
+* only_integer
+* greater_than - Specifies the value must be greater than the supplied value. The default error message for this option is "must be greater than %{count}".
+* greater_than_or_equal_to - Specifies the value must be greater than or equal to the supplied value. The default error message for this option is "must be greater than or equal to %{count}".
+* equal_to - Specifies the value must be equal to the supplied value. The default error message for this option is "must be equal to %{count}".
+* less_than - Specifies the value must be less than the supplied value. The default error message for this option is "must be less than %{count}".
+* less_than_or_equal_to - Specifies the value must be less than or equal to the supplied value. The default error message for this option is "must be less than or equal to %{count}".
+* other_than - Specifies the value must be other than the supplied value. The default error message for this option is "must be other than %{count}".
+* odd - Specifies the value must be an odd number if set to true. The default error message for this option is "must be odd".
+* even - Specifies the value must be an even number if set to true. The default error message for this option is "must be even".
+
+## Conditional Validation
+
+It is possible to validate attribute based on the state of other attribute from the form. 
+{% raw %}
+<pre><code class="language-yaml">
+            expiry_year:
+              validation:
+                presence:
+                  if: '{% if form.expires %}true{% endif %}'
+</code></pre>
+{% endraw %}
+
+
+## Common Validation Options
+
+### message
+It is possible to overwrite the default message with additional configuration:
+
+{% raw %}
+<pre><code class="language-yaml">
+      customizations:
+        expertise:
+          validation:
+            presence:
+              message: 'record is required.'
+</code></pre>
+{% endraw %}
+
+### allow_nil
+The ```allow_nil``` option skips the validation when the value being validated is nil.
+
+### allow_blank
+Allows to pass an empty string.
+
+
+### Validation Errors
+
+When form validation is raised, error message is added to ```form.errors``` object that contains all attribute - error information. 
+In your liquid view you can itterate over that object as in the example below:
+{% raw %}
+<pre><code class="language-yaml">
+      {% for error in form.errors %}
+        {{ error[0] }} {{ error[1] }}
+      {% endfor %}
+</code></pre>
+{% endraw %}
+
+Errors can be fetched throgh the form_object as well, this is very useful if you want to display errors next to the input field:
+{% raw %}
+<pre><code class="language-yaml">
+form_object.object.errors[field_name]
```

Comment:
I think form_object is not something we are promoting right now - errors should be taken from form_builder - correct me if im wrong @Slashek